connector/src/main/scala/com/basho/riak/spark/rdd/partitioner/RiakTSPartitioner.scala

```diff
@@ -17,27 +17,175 @@
+          s"It's not a valid filter $filter to be pushed down, only >, <, >=, <= and =  are allowed.")
+    }
+
+    // TODO: need to add pattern matching for values, to be sure that they are used correctly
+    (s"$attribute $sqlOperator ?", value)
+  }
+
+  private def whereClause(filters: Array[Filter]): (String, Seq[Any]) = {
+    val sqlValue = filters.map(filterToSqlAndValue)
+    val sql = sqlValue.map(_._1).mkString(" AND ")
+    val args = sqlValue.map(_._2)
+    (sql, args.seq)
+  }
+
+  private def getMin(filters: Array[Filter]): Option[Long] = {
+    filters.flatMap {
+      case GreaterThanOrEqual(attr, v: Long)      => Some(v)
+      case GreaterThan(attr, v: Long)             => Some(v + 1)
+      case GreaterThanOrEqual(attr, v: Timestamp) => Some(v.getTime)
+      case GreaterThan(attr, v: Timestamp)        => Some(v.getTime + 1)
+      case _                                      => None
+    }.headOption
+  }
+
+  private def getMax(filters: Array[Filter]): Option[Long] = {
+    filters.flatMap {
+      case LessThanOrEqual(attr, v: Long)      => Some(v + 1)
+      case LessThan(attr, v: Long)             => Some(v)
+      case LessThanOrEqual(attr, v: Timestamp) => Some(v.getTime + 1)
+      case LessThan(attr, v: Timestamp)        => Some(v.getTime)
+      case _                                   => None
+    }.headOption
+  }
+
+  private def getTsFieldName(readConf: ReadConf, schema: Option[StructType]): Option[String] = {
+    if (readConf.tsFieldName != null) {
+      Some(readConf.tsFieldName)
+    } else {
+      schema match {
+        case Some(s) => s.fields.filter(_.dataType == TimestampType).map(_.name).headOption
+        case None    => None
+      }
+    }
+  }
+
+  def singlePartition(endpoints: Iterable[HostAndPort], bucketName: String, schema: Option[StructType],
+                      columnNames: Option[Seq[String]], query: Option[String], whereConstraints: Option[(String, Seq[Any])]): Array[Partition] = {
+    val (sql, values) = query match {
+      case None => toSql(columnNames, bucketName, schema, whereConstraints)
+      case Some(q) => (q, Seq.empty[Nothing])
+    }
+    Array(new RiakTSPartition(0, endpoints, TSQueryData(sql, values)))
+  }
+
+  def partitions(endpoints: Iterable[HostAndPort], bucketName: String, schema: Option[StructType],
+                 columnNames: Option[Seq[String]], filters: Array[Filter], readConf: ReadConf): Array[Partition] = {
+    val splitCount = readConf.splitCount
+
+    val filtersByAttr = {
+      filters
+        .map(f => (getFilterAttribute(f), f))
+        .filter(f => f._1 != null)
+        .groupBy(attrFilter => attrFilter._1)
+        .mapValues(a => a.map(p => p._2))
+    }
+
+    val tsFieldName = getTsFieldName(readConf, schema) match {
+      case Some(name) => name
+      case None       => null
+    }
+    
+    if(tsFieldName == null) {
+      return singlePartition(endpoints, bucketName, schema, columnNames, None, Some(whereClause(filters)))
```

Comment:
It worth at least to be mentioned in the log to warn user about the case when SC can't split data as uxpected